netlify-cli: 6.13.2 -> 12.2.4, esbuild_netlify: 0.13.6 -> 0.14.39

panda2134 opened the pull request:
Signed-off-by: panda2134 <ljypanda@live.com>

###### Description of changes

netlify-cli: 6.13.2 -> 12.2.4
esbuild_netlify: 0.13.6 -> 0.14.39 (dependency of netlify-cli written in Golang)

Changelog: https://github.com/netlify/cli/releases

###### Things done

I only tried building on x86_64-darwin, but it should work on other platforms.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested (with old test cases)
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

panda2134 committed:
esbuild_netlify: fix trailing space in esbuild_netlify

panda2134 committed:
netlify-cli: update release note for updating netlify-cli

Kranzes commented:
Result of `nixpkgs-review pr 201937` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>esbuild_netlify</li>
    <li>netlify-cli</li>
  </ul>
</details>

panda2134 commented:
@cdepillabout Hi, may I ask what shall I do to get this merged? I understand there're plenty of PRs, and you're quite busy merging them. Hence I wonder what I can do on my side. (btw enabling sandbox on macOS seems to be impossible?)

panda2134 commented on pkgs/development/web/netlify-cli/generate.sh:
```diff
@@ -2,13 +2,13 @@
   --composition composition.nix \
   --node-env node-env.nix \
-  --nodejs-14 \
+  --nodejs-16 \
```
I bumped NodeJS dependency since Node14 is getting EOL soon. Maybe we can switch to Node18?

jtojnar commented on pkgs/development/web/netlify-cli/generate.sh:
```diff
@@ -2,13 +2,13 @@
 cd "$( dirname "${BASH_SOURCE[0]}" )"
 rm -f ./node-env.nix
-src="$(nix-build --expr '(import ../../../.. {}).fetchFromGitHub (lib.importJSON ./netlify-cli.json)')"
+src="$(nix-build --expr 'let pkgs = import ../../../.. {}; lib = import ../../../../lib; meta = (lib.importJSON ./netlify-cli.json); in pkgs.fetchFromGitHub {owner = meta.owner; repo = meta.repo; rev = meta.rev; sha256 = meta.sha256;}')"
```
You can use `pkgs.lib` instead of a separate import. And `inherit (meta) owner repo rev sha256;`.

panda2134 replied:
okay, i'll change accordingly

roberth commented:
Please adhere to the contributing guidelines. That way ofborg will pick up on the change packages, based on the commit message prefix.

SuperSandro2000 commented on pkgs/development/tools/esbuild/netlify.nix:
```diff
@@ -7,28 +7,16 @@
 buildGoModule rec {
   pname = "esbuild";
-  version = "0.13.6";
+  version = "0.14.39";
```
Why not https://github.com/evanw/esbuild/releases/tag/v0.15.14 ?

panda2134 replied:
That's because netlify-cli requires a fork of the original esbuild with new features. The link you provided is the original esbuild codebase.

panda2134 commented:
> Please adhere to the contributing guidelines. That way ofborg will pick up on the change packages, based on the commit message prefix.

Thank you for your notice, but how can I mention bumping two packages in the title?

panda2134 committed:
netlify-cli: refactor

Co-authored-by: Robert Hensing <roberth@users.noreply.github.com>

roberth commented:
> how can I mention bumping two packages in the title?

It doesn't have to be in the title of the pull request, just in the commits. I'm not actually sure whether ofborg can pick up multiple attributes from a single commit message, but you have four commits, so that helps.

You could split up [bump: esbuild_netlify to 0.14.39 and netlify-cli to 12.2.4](https://github.com/NixOS/nixpkgs/pull/201937/commits/fe6b478ea30725d0faf25169872b28b0b2e88c99) into two commits, but `bump` is the wrong package ;)

Maybe you're already familiar with the rebase command. Otherwise, I can recommend jonringer's video https://www.youtube.com/watch?v=OR8Q-4PLalc

Please read the examples in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes and use the imperative mood in the commit messages.

panda2134 commented:
> > how can I mention bumping two packages in the title?
> 
> It doesn't have to be in the title of the pull request, just in the commits. I'm not actually sure whether ofborg can pick up multiple attributes from a single commit message, but you have four commits, so that helps.
> 
> You could split up [bump: esbuild_netlify to 0.14.39 and netlify-cli to 12.2.4](https://github.com/NixOS/nixpkgs/pull/201937/commits/fe6b478ea30725d0faf25169872b28b0b2e88c99) into two commits, but `bump` is the wrong package ;)
> 
> Maybe you're already familiar with the rebase command. Otherwise, I can recommend jonringer's video https://www.youtube.com/watch?v=OR8Q-4PLalc
> 
> Please read the examples in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes and use the imperative mood in the commit messages.

Thanks a lot for your help. I've renamed the commits accordingly to reflect changes in the two packages.

panda2134 committed:
esbuild_netlify: 0.13.6 -> 0.14.39

This is the esbuild fork used by netlify-cli. The commit
updates it to its latest version. Note that versioning of
this fork is different from evanw/esbuild.

panda2134 committed:
netlify-cli: 6.13.2 -> 12.2.4

Updates netlify-cli to 12.2.4, while switching to NodeJS 16.
According to its issue page, there seems to be problems running
this tool with NodeJS 18, so we'd better stick to NodeJS 16
instead.

roberth commented on pkgs/development/web/netlify-cli/default.nix:
```diff
@@ -1,12 +1,15 @@
 { callPackage, fetchFromGitHub, lib, pkgs }:
 let
   nodePackages = import ./composition.nix { inherit pkgs; };
+  meta = (lib.importJSON ./netlify-cli.json);
```
```suggestion
  sourceInfo = (lib.importJSON ./netlify-cli.json);
```

\+ below

`meta` already has a special meaning in packages.

roberth commented:
Builds on darwin, `--help` works on darwin, but I didn't get the dev server `passthru.tests` test to work yet. It wants to reverse proxy to a supposed web application server at port 3999, which doesn't exist. This may be a problem with the test itself, although it doesn't appear on linux.

panda2134 commented:
> It wants to reverse proxy to a supposed web application server at port 3999, which doesn't exist. This may be a problem with the test itself, although it doesn't appear on linux.

It might be related to the firewall on MacOS, which pops up a warning and closes it in half a second. This is what happened on my Mac.

panda2134 commented:
> EDIT: Not inclined. Darwin patches are deleted; starting my own build now.

A manual test shows the reverse proxy to be working after clicking YES on firewall warning. Tested on my machine.

roberth commented:
@panda2134 that explains a lot. I was using an ssh session, so I didn't see it. Which port did you allow? Is this necessary for listening on localhost, or only when listening on network-accessible addresses?

panda2134 commented:
> Which port did you allow?

I didn't try to allow a specific port, as the popup was quite vague, asking me to allow "incoming connections" for netlify-cli.

> Is this necessary for listening on localhost, or only when listening on network-accessible addresses?

I guess both, considering the message given in the popup window.

panda2134 commented:
> @panda2134 that explains a lot. I was using an ssh session, so I didn't see it. Which port did you allow? Is this necessary for listening on localhost, or only when listening on network-accessible addresses?

Alternatively you can try turning off the firewall entirely (in System Preferences) and see if the tests pass.

roberth commented on pkgs/development/web/netlify-cli/default.nix:
```diff
@@ -1,12 +1,15 @@
     '';
-    src = fetchFromGitHub (lib.importJSON ./netlify-cli.json);
+    src = fetchFromGitHub {
+      inherit (meta) owner repo rev sha256;
```
```suggestion
      inherit (sourceInfo) owner repo rev sha256;
```

roberth commented:
Part of the test problem on darwin may be that `host localhost` fails in the sandbox, causing netlify to bind on all interfaces, but I haven't been able to confirm that yet.

roberth committed:
netlify-cli.tests.test: Add ps on darwin
